cou/apps/auxiliary.py

```diff
@@ -91,3 +91,33 @@ def channel_codename(self) -> OpenStackRelease:
                 f"for channel: '{self.channel}'"
             )
         )
+
+
+@AppFactory.register_application(["mysql-innodb-cluster"])
+class MysqlInnodbClusterApplication(OpenStackAuxiliaryApplication):
+    """Application for mysql-innodb-cluster charm."""
+
+    def _get_upgrade_current_release_packages_plan(self, parallel: bool = False) -> UpgradeStep:
+        """Get Plan for upgrading software packages to the latest of the current release.
+
+        :param parallel: Parallel running, defaults to False
+        :type parallel: bool, optional
+        :return: Plan for upgrading software packages to the latest of the current release.
+        :rtype: UpgradeStep
+        """
+        description = (
+            f"Upgrade software packages of '{self.name}' from the current APT repositories"
+        )
+
+        # NOTE(agileshaw): holding 'mysql-server-core-8.0' package prevents undesired
+        # mysqld processes from restarting, which lead to outages
+        packages_to_hold = ["mysql-server-core-8.0"]
```

Comment:
I'm wondering if this will happen on other applications. If this is likely to happen, I would prefer having `packages_to_hold` as a class attribute and not overriding the `_get_upgrade_current_release_packages_plan`.

Reply:
So far this is the only application we need to do packages holding. However, I do want to make it as generic as possible for future proofing. Thanks for the suggestion, I've updated the code.